mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -1188,7 +1190,7 @@ def EmitC_AssignOp : EmitC_Op<"assign", []> {
 }
 
 def EmitC_YieldOp : EmitC_Op<"yield",
-      [Pure, Terminator, ParentOneOf<["ExpressionOp", "IfOp", "ForOp"]>]> {
+      [Pure, Terminator, ParentOneOf<["ExpressionOp", "IfOp", "ForOp", "SwitchOp"]>]> {
```

Comment:
With this change you need to update https://github.com/llvm/llvm-project/blob/b9793a6f04b3ab1ff9ed86a675c93e957ec52f51/mlir/test/Dialect/EmitC/invalid_ops.mlir#L230 accordingly. The builtbot fails, please see the logs!